inference-engine/tests/ie_test_utils/common_test_utils/ngraph_test_utils.cpp

```diff
@@ -83,6 +85,511 @@ std::string name(const Node& n) {
+    }
+    template <typename AttrValue>
+    AttributeStorage<AttrValue>& storage() {
+        return *static_cast<AttributeStorage<AttrValue>*>(this);
+    }
+
+    size_t stored_attributes_number() const {
+        return storage<MemoryChunk>().get_attributes_number() +
+               storage<bool>().get_attributes_number() +
+               storage<std::string>().get_attributes_number() +
+               storage<int8_t>().get_attributes_number() +
+               storage<int16_t>().get_attributes_number() +
+               storage<int32_t>().get_attributes_number() +
+               storage<int64_t>().get_attributes_number() +
+               storage<uint8_t>().get_attributes_number() +
+               storage<uint16_t>().get_attributes_number() +
+               storage<uint32_t>().get_attributes_number() +
+               storage<uint64_t>().get_attributes_number() +
+               storage<float>().get_attributes_number() +
+               storage<double>().get_attributes_number() +
+               storage<std::vector<int8_t>>().get_attributes_number() +
+               storage<std::vector<int16_t>>().get_attributes_number() +
+               storage<std::vector<int32_t>>().get_attributes_number() +
+               storage<std::vector<int64_t>>().get_attributes_number() +
+               storage<std::vector<uint8_t>>().get_attributes_number() +
+               storage<std::vector<uint16_t>>().get_attributes_number() +
+               storage<std::vector<uint32_t>>().get_attributes_number() +
+               storage<std::vector<uint64_t>>().get_attributes_number() +
+               storage<std::vector<float>>().get_attributes_number() +
+               storage<std::vector<double>>().get_attributes_number() +
+               storage<std::vector<std::string>>().get_attributes_number() +
+               storage<SubGraphOpInputDescription>().get_attributes_number() +
+               storage<SubGraphOpOutputDescription>().get_attributes_number() +
+               storage<SpecialBodyPorts>().get_attributes_number();
+    }
+};
+
+}  // namespace storage
+
+class ReadAndStoreAttributes : public ngraph::AttributeVisitor, protected storage::Storage {
+public:
+    void on_adapter(const std::string& name, ngraph::ValueAccessor<void>& adapter) override {
+        if (auto inputs =
+                ngraph::as_type<ngraph::AttributeAdapter<SubGraphOpInputDescription>>(&adapter)) {
+            insert(name, inputs->get());
+        } else if (
+            auto outputs =
+                ngraph::as_type<ngraph::AttributeAdapter<SubGraphOpOutputDescription>>(&adapter)) {
+            insert(name, outputs->get());
+        } else if (
+            auto ports = ngraph::as_type<ngraph::AttributeAdapter<SpecialBodyPorts>>(&adapter)) {
+            insert(name, ports->get());
+        } else {
+            std::cout << "store   attr [ ERR ]: " << name << " [drop `void` comparison which is '"
```

Comment:
I think we should throw here to fail the test and indicate that there is unhandled attribute. We'll then get a bug report and we'll be able to fix it. If we ignore it silently it will look like comparison succeeded.

Reply:
OK. Good idea.